src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -1277,8 +1311,18 @@ class spell_pri_prayer_of_mending_dummy : public spell_pri_prayer_of_mending_Spe
 
     void HandleEffectDummy(SpellEffIndex /*effIndex*/)
     {
+        Unit* caster = GetCaster();
+        Unit* target = GetHitUnit();
+
         // Note: we need to increase BasePoints by 1 since it's 4 as default.
-        CastPrayerOfMendingAura(GetCaster(), GetHitUnit(), GetEffectValue() + 1, true);
+        CastPrayerOfMendingAura(caster, target, GetEffectValue() + 1, true);
+
+        // Note: Epiphany talent. Only the non-triggered version should consume it.
+        if (caster->HasAura(SPELL_PRIEST_EPIPHANY) && m_scriptSpellId == SPELL_PRIEST_PRAYER_OF_MENDING)
+        {
+            if (caster->HasAura(SPELL_PRIEST_EPIPHANY_HIGHLIGHT))
+                caster->RemoveAurasDueToSpell(SPELL_PRIEST_EPIPHANY_HIGHLIGHT);
```

Comment:
you can just do RemoveAura(SPELL_PRIEST_EPIPHANY_HIGHLIGHT) without checking before if the Aura is applied.. just simple Remove